game/action.go

```diff
@@ -7,10 +7,13 @@ import (
 type Action struct {
 	Type ActionType
 
+	// a faux effect that resolves after a choice-based action, such as returning Scry cards and drawing
+	AfterEffect                     *Effect
 	Card                            *Card
 	EntersTheBattleFieldSpellTarget *StackObject // the spell target Card's coming into play effect
 	Cost                            *Cost
 	Selected                        []*Permanent // for non-targetted effects, such as in Snap
+    ShouldSwitchPriority            bool // whether to switch priority after the action
```

Comment:
what is ShouldSwitchPriority all about, that doesn't seem like a thing that is supposed to exist